designcompose/src/main/java/com/android/designcompose/Layout.kt

```diff
@@ -56,11 +56,36 @@ import java.util.Optional
 import kotlin.math.roundToInt
 
 internal data class TextMeasureData(
+    val textHash: Int,
     val paragraph: ParagraphIntrinsics,
     val density: Density,
     val maxLines: Int,
     val autoWidth: Boolean,
-)
+) {
```

Comment:
This stuff is all optimization; hashing TextMeasureData was number 1 in the profile for a while.